### DISCUSSION TITLE
Questions about assembly with merged paired-end reads

### DISCUSSION THREAD

**Discussion by darksky1605** [ablab/spades#765] [Q&A] [answered]
Hello, could you tell me if there are constraints for files which can be passed via `--merged` parameter?

Can merged reads extend outwards, too, i.e. additional extension to the left of read 1 and to the right of read 2? Or does SPAdes expect that the merged reads start with and end with one of the reads of the merged read pair, respectively?

What are the benefits of using `--merged` instead of specifying them as single-end reads via `-s`?

Should the paired-end file  `(--12)` which is used together with the merged file also contain the original read pairs which are merged, or not?

**Comment by asl** [comment, marked as answer]
Hello!

SPAdes expects that merged reads directly correspond to the original paired-end reads. In many cases is treats them as paired-end one (e.g. for the purposes of repeat resolution). In order to correctly estimate the original fragment length distribution the unmerged part should be provided as paired-end part of a sequencing library.

Also, using merged reads could allow one to increase the maximum k-mer length in some cases (e.g. merged 2x100 bp reads could allow k-mer lengths up to 77 like 2x150 bp). Some additional information also could be found here: https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.102

**Reply by schorlton** [reply]
Hi @asl , thanks for this information. To help me understand:
If I have reads `a_1`, `a_2`, `b_1` and `b_2`. If `a` are merged into `a_merged`, and `b` are unmerged, I expected to submit:
`-1 b_1 -2 b_2 --merged a_merged`. This is based on this section of the docs:
" Non-empty files with (_remaining_) unmerged left/right reads (separate or interlaced) must be provided for the same library for SPAdes to correctly detect the original read length."

However, what I understand from [this post](https://github.com/ablab/spades/discussions/751) (I recognize things may have changed) and your message is possibly my command should look like:
`-1 a_1+b_1 -2 a_2+b+2 --merged a_merged`?

Additionally, what should I do if there is also `c_1` and `c_2`, but `c_2` gets filtered? I tried specifying `-1 b_1 -2 b_2 --merged a_merged --single c_1` but it appears SPAdes does not support this in meta mode. Can I add the singletons to the merged? Thanks!

**Reply by rohansachdeva** [reply]
@asl Clarification on this would be helpful. Should the paired end input -1 and -2 include all of the paired reads or just the reads that could not be merged?

**Reply by asl** [reply]
The manual (https://cab.spbu.ru/files/release3.15.5/manual.html) says:

```
If you have merged some of the reads from your paired-end (not mate-pair or high-quality mate-pair) library, you should provide the file with resulting reads as a "merged read file" for the corresponding library.
Note that non-empty files with the remaining unmerged left/right reads (separate or interlaced) must be provided for the same library (for SPAdes to correctly detect the original read length).
```

**Reply by rohansachdeva** [reply]
Thanks for the reply - I think the confusion comes from: https://github.com/ablab/spades/discussions/751#discussioncomment-733956

Where you state: "For your merged reads you need to provide also the original unmerged reads"

Which I read as we should use all of the paired end reads where as the manual says to use only the paired reads that cannot be merged.
